ska_dbi/sqsh.py

```diff
@@ -0,0 +1,168 @@
+import subprocess
+from pathlib import Path
+from astropy.table import Table
+from ska_dbi.common import DEFAULT_CONFIG, NoPasswordError
+
+SYBASE = "/soft/SYBASE16.0"
+ENV_VARS = [
+    f"SYBASE={SYBASE}",
+    "LD_LIBRARY_PATH=/soft/SYBASE16.0/OCS-16_0/lib:/soft/SYBASE16.0/OCS-16_0/lib3p64:/soft/SYBASE16.0/OCS-16_0/lib3p",
+]
+SQSH_BIN = "/usr/local/bin/sqsh.bin"
+
+
+class Sqsh(object):
+    """
+    Class to interface with sqsh on HEAD to query the axafapstat or axafocat databases.
+
+    Example usage::
+
+      db = DBI(dbi='sqlite', server=dbfile, numpy=False, verbose=True)
+      db = DBI(dbi='sybase', server='sybase', user='aca_ops', database='aca')
+      db = DBI(dbi='sybase')   # Use defaults (same as above)
+
+    :param server: Server name (default = sqlsao)
+    :param user: User name (default = aca_ops)
+    :param passwd: Password (optional).  Read from aspect authorization dir if required and not supplied.
+    :param database: Database name (default = axafapstat)
+    :param sqshrc: sqshrc file (optional).  Read from aspect authorization dir if required and not supplied.
+    :param authdir: Directory containing authorization files
+
+    :rtype: Sqsh object
+    """
+    def __init__(
+        self,
+        server=None,
+        user=None,
+        passwd=None,
+        database=None,
+        sqshrc=None,
+        authdir="/proj/sot/ska/data/aspect_authorization",
+        **kwargs,
+    ):
+
+        self.server = server or DEFAULT_CONFIG['sybase'].get("server")
+        self.user = user or DEFAULT_CONFIG['sybase'].get("user")
+        self.database = database or DEFAULT_CONFIG['sybase'].get("database")
+        self.passwd = passwd
+        self.sqshrc = sqshrc
+
+        if not Path(SYBASE).exists():
+            raise RuntimeError(f"SYBASE does not exist: {SYBASE}")
+        if not Path(SQSH_BIN).exists():
+            raise RuntimeError(f"sqsh.bin does not exist: {SQSH_BIN}")
+
+        if self.passwd is None and self.sqshrc is None:
+            sqshrc_file = (
+                Path(authdir) / f"sqsh-{self.server}-{self.database}-{self.user}"
+            )
+            if sqshrc_file.exists():
+                self.sqshrc = sqshrc_file
+            else:
+                raise NoPasswordError(
+                    f"None supplied and unable to read inferred sqshrc file {sqshrc_file}"
+                )
+
+    def __enter__(self):
+        """Context manager enter runtime context.  No action required, just return self."""
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """
+        Context manager exit run time context.
+        """
+        pass
```

Comment:
You can remove that `pass`. The docstring is enough to satisfy the syntax of an empty function. Or if you really want something then `...` is more indicative of doing nothing here.